llama_index/vector_stores/google/generativeai/genai_extension.py

```diff
@@ -137,20 +137,32 @@ class Config:
         user_agent: The user agent to use for logging.
         page_size: For paging RPCs, how many entities to return per RPC.
         testing: Are the unit tests running?
+        google_auth_credentials: For setting credentials such as using service accounts
     """
 
     api_endpoint: str = _DEFAULT_API_ENDPOINT
     user_agent: str = _USER_AGENT
     page_size: int = _DEFAULT_PAGE_SIZE
     testing: bool = False
+    google_auth_credentials: Optional[credentials.Credentials] = None
 
+    @classmethod
+    def with_credentials(cls, auth_creds: credentials.Credentials) -> "Config":
+        return cls(
+            google_auth_credentials=auth_creds,
+        )
```

Comment:
Because this is a dataclass already, we don't need this class method. The following alone will work:

    Config(credentials=<whatever>)

Note that I've also simplified `google_auth_credentials` to just `credentials`. Less verbose that way.

Reply:
ah thanks!